60 orders screen edit orderproduct

sspoon865 opened the pull request:
Closes #59 and closes #60

A LOT was added by this.

- Added Tabs to Order Add/Edit screen
- Order / Products tab (Products starts disabled if adding new order, then gets redirected once order is added)
- Added products grid to Add/Edit screen
- Added Add/Edit product modal
- Added all the queries to support adding and editing products to an order (so many queries...)

Some outstanding **issues**:
- Removing a product from an order on the Web Portal does not remove it from orderProducts GraphQL query. Will put an issue in on scale-backend (probably re-open [#12](https://github.com/TAC-Insight/fw-scale-backend/issues/12)).
- In current desktop app, you can delete a product from an order. If we still want to allow for this, the mutation is not yet available for it.
- With how a user is redirected to the "Products" tab after creating an order, refreshing the page will not show the same order but instead the "New Order" screen. (This is the same behavior as the Web Portal, but it could be possible to replace the History entry in the browser with params to keep the user on the same order when refreshing—downside to that is there may be some funky behavior with back button in browser, but would need to test to confirm.)

HaleyBoles commented:
Error from??? when adding a new Product to Order #9
![image](https://github.com/TAC-Insight/fw-scale-frontend/assets/128633188/c11723a7-d2cc-449b-99bf-8e19f3c2fb61)
code: 
"UNABLE_TO_SAVE_ORDER"
message: 
"Portal\nWashington String or binary data would be truncated.\r\nThe statement has been terminated. Error Code: Washington\n{\"Message\":\"String or binary data would be truncated.\\r\\nThe statement has been terminated. Error Code: Washington\",\"ErrorCode\":\"Washington\",\"StatusCode\":409}"
serverMessage: 
"Portal\nWashington String or binary data would be truncated.\r\nThe statement has been terminated. Error Code: Washington\n{\"Message\":\"String or binary data would be truncated.\\r\\nThe statement has been terminated. Error Code: Washington\",\"ErrorCode\":\"Washington\",\"StatusCode\":409}"
__typename: 
"TacServerError"
![image](https://github.com/TAC-Insight/fw-scale-frontend/assets/128633188/e4600375-e425-4c60-a4b3-0b2d390ee6af)

HaleyBoles commented:
Also may be worth considering including Order Quantity on Product add/edit. While this is not in the current app, would probably be a nice-to-have.

ericwilhite commented on src/routes/OrdersAddEdit.tsx:
```diff
@@ -141,321 +324,1199 @@ export function OrdersAddEditRoute() {
+											/>
+										</div>
+										<Divider />
+										<div className="flex flex-col space-y-4 px-1 overflow-scroll">
```
`overflow-scroll` is causing the scroll bars to always show

`overflow-auto` would show them only when necessary. And I don't think we want horizontal scrolling here. So I would probably just do `overflow-y-auto`

ericwilhite reviewed (changes requested):
Added and updated a product and worked great!

See overflow note

Also, I do worry that the descriptions are making it "busy" and many of them are unnecessary. Ex: Tax Code: Enter the tax code. Freight Rate: Enter the freight rate. 

Not sure what it would look like if some have and some don't have them. Conflicted.

sspoon865 commented:
Broke out adding/editing orderProducts to its own page.
Added character limits to forms.
Added Haul Zone checks.
Minor fixes to Orders screen (affected by query changes).

* ~Removing a product from an order on the Web Portal does not remove it from orderProducts GraphQL query.~ **Fixed.** ✅
* In current desktop app, you can delete a product from an order. If we still want to allow for this, the mutation is not yet available for it.
* ~With how a user is redirected to the "Products" tab after creating an order, refreshing the page will not show the same order but instead the "New Order" screen.~ **Fixed.** ✅

ericwilhite reviewed (approved):
I'm approving this because it builds and YOLO

Need to do a lot of testing and we can break any changes up into follow-up PRs
